flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/split/DynamoDbStreamsShardSplitSerializer.java

```diff
@@ -63,6 +65,11 @@ public byte[] serialize(DynamoDbStreamsShardSplit split) throws IOException {
                     out.writeUTF((String) startingMarker);
                 }
             }
+            out.writeInt(split.getParentShardIds().size());
+            for (String parentShardId : split.getParentShardIds()) {
+                out.writeUTF(parentShardId);
+            }
```

Comment:
can be simplified once we remove the set of `parentShardIds`

Reply:
changed